portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,97 +70,88 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       
-      // TODO: 
-      //      Get default tag from the InternalTags class and use that below.
+      //TODO: Handle image file sent with FormData.
+      //String imageUrl = getUploadedFileUrl(request, imagePath);
       
-      // Get type of comment.
-      String commentType = getParameter(request, "tags", "Default");
-
-      //String imageUrl = getUploadedFileUrl(request, "image");
+      // Only put BlogMessages with a message and tag in datastore.
+      if (message.equals("") || message.equals(null) || postTag == null) {
+        return;
+      }
 
-      String messageRepliesString = getParameter(request, "replies", "");
-	    String messageRepliesArray[] = messageRepliesString.split(",");
-	    List<String> messageReplies = new ArrayList<String>();
-	    messageReplies = Arrays.asList(messageRepliesArray);
-      
       long timestamp = System.currentTimeMillis();
-
       Entity blogMessageEntity = new Entity("blogMessage");
       blogMessageEntity.setProperty("sender", sender);
       blogMessageEntity.setProperty("text", message);
       //blogMessageEntity.setProperty("imgUrl", imageUrl);
       blogMessageEntity.setProperty("time", timestamp);
-      blogMessageEntity.setProperty("tag", commentType);
+      blogMessageEntity.setProperty("tag", postTag);
       blogMessageEntity.setProperty("replies", messageReplies);
+    
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(blogMessageEntity);
 
-      response.sendRedirect("/index.html");
+      // To get the recently posted message, add its tag to the |tagsToSearch| list.
+      // If the list has tags already, clear them before adding the tag.
+      // Our goal is to get the recently posted message.
+      if (!tagsToSearch.isEmpty()) {
+        tagsToSearch.clear();
+      }
+      tagsToSearch.add(postTag);
+
+      // Everytime a post comes through, the javascript function that handles the Post assigns
+      // a load_factor value of 1.
+      // We assign that to the |numberOfCommentsToDisplay|. 
+      if (loadFactor != null) {
+        // we don't catch since this is always 1.
+        numberOfCommentsToDisplay = Integer.parseInt(loadFactor); 
+      }
+      
+      // Now we call |doGet| to load and respond with the message.
+      // |doGet| passes the |tagsToSearch| and |numberOfCommentsToDisplay| to the 
+      // BlogHashMap's getMessages method, which responds with the recent post.
+      doGet(request, response);
     }
 
-  /**
-   * @return the request parameter, or the default value if the parameter
-   *         was not specified by the client.
-   */
-    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    // Returns the requested parameter, or the default value if the parameter is null.
+    private String getTagParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Why do we need this change instead of reusing getParameter()? getParameter() sounds more general.

Reply:
In our case, this function is only useful for the getting the tag because if a user fails to include a tag we want to assign the defautTag. We can't assign default values for a user's name or message.
_**I should have changed the parameter names to sth like (...request, String Tag, String defaultTag) to make it's functionality more obvious.**_